### PR TITLE
Register types for specific Kotlin calls

### PR DESCRIPTION
**Pull request by ursachec**

### CODE REVIEW

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala [hunk 1]
```diff
@@ -401,6 +401,16 @@ class DefaultTypeInfoProvider(environment: KotlinCoreEnvironment) extends TypeIn
     }
   }
 
+  def containingDeclFullName(expr: KtCallExpression): Option[String] = {
+    val resolvedDesc = resolvedCallDescriptor(expr)
+    resolvedDesc match {
```
Comment (note) [inline joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala:406]: Could be rewritten as:

```
resolvedDesc.map { fnDescriptor =>
  val decl = fnDescriptor.getContainingDeclaration
  TypeRenderer.renderFqName(decl)
}
```

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala [hunk 2]
```diff
@@ -2612,6 +2612,11 @@ class AstCreator(fileWithMeta: KtFileWithMeta, xTypeInfoProvider: TypeInfoProvid
     fileInfo: FileInfo,
     typeInfoProvider: TypeInfoProvider
   ): AstWithCtx = {
+    val declFullNameOption = typeInfoProvider.containingDeclFullName(expr)
+    if (declFullNameOption.isDefined) {
```
Comment (note) [inline joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala:2616]: ```declFullNameOption.foreach(registerType)```

Comment (ursachec) [reply joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala:2616]: sweet, thanks!

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala (hunk 1 above)
Comment (ursachec) [reply joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala]: cool, thx

**Review by max-leuthaeuser — APPROVED** [review]

**Review by mpollmeier — APPROVED** [review]

**Pull request merged**